Add support for specifying SNAT source and destination filters

priteau opened the pull request:
This is useful if forwarded packets need to exit on a different
interface depending on the source or destination IP address or port.

Change-Id: Ifbfbade4baaa1901b08549e52acc725e45379a16
(cherry picked from commit 3903ca92e92288e026d5923ab9a7c4d65d2e2d4a)

markgoddard commented:
Generated docs are available here: http://kayobe-personal.readthedocs.io/en/networking-docs/configuration/network.html

markgoddard commented:
Last change should have fixed doc8 failure.

priteau committed:
Add support for specifying SNAT source and destination filters

This is useful if forwarded packets need to exit on a different
interface depending on the source or destination IP address or port.

Change-Id: Ifbfbade4baaa1901b08549e52acc725e45379a16
(cherry picked from commit 3903ca92e92288e026d5923ab9a7c4d65d2e2d4a)
